Slim/Control/Queries.pm

```diff
@@ -5010,6 +5011,22 @@ sub _songData {
 			$remoteMeta->{T} = $remoteMeta->{samplerate};
 			$remoteMeta->{I} = $remoteMeta->{samplesize};
 			$remoteMeta->{W} = $remoteMeta->{releasetype};
+
+			$remoteMeta->{V} = -1;
+            # Distance from the live edge of live remote stream. -1 is not live, 0 is live at the edge, >0 is distance in seconds from the live edge.
+			if ( my $client = $request->client ) {
+				if (my $song = $client->currentSongForUrl($url)) {
+					if ( $handler->can('isLive') )  {
+						if ( $handler->isLive($song) ) {# Distance form the live edge, if available.
+							$remoteMeta->{V} = $remoteMeta->{live_edge} ? $remoteMeta->{live_edge} : 0;
+						} else { # Not Live
+							$remoteMeta->{V} = -1;
+						}
+					} else { # only live if the song has isLive set
+						$remoteMeta->{V} = $song->isLive() ? 0 : -1;
```

Comment:
Wouldn't you set that as the default value and override it only if the handler has something to say? Feels easier to read to me.

Reply:
Yes, fair point.